test/test_helper.exs

```diff
@@ -1,10 +1,7 @@
-Mox.defmock(KinesisClient.KinesisMock, for: KinesisClient.Kinesis.Adapter)
-Mox.defmock(KinesisClient.Stream.AppStateMock, for: KinesisClient.Stream.AppState.Adapter)
```

Comment:
Since these were previously the only `Mox` mocks, and this PR adds two more only to then copy all four into `Mimic`, what do you think about just replacing Mox with Mimic altogether? 😅 

Just an idea to remove a dependency in favor of one we're already adding. 🙏

Reply:
Ah, yeah the test suite is a bit of a mess right now.  I need to fix up the mocks for sure 😄